### ISSUE TITLE
Add task using Celery to send email verification for reset password

### ISSUE THREAD

**Issue by keshakaneria** [anitab-org/vms#1030]
## Description
In order to login into the system using forgot password, I need to verify my email address to get logged in to change the old password with the new one. Currently, the email is displayed in the console itself.

### Update [Required]
- [ ] Add tasks using Celery
- [ ] Email should be sent to the email address instead of printing it in console.

Celery is an asynchronous task scheduler which helps to run different process in background which can be used here in order to create tasks for mailing system.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
3 hours

**Comment by SanketDG** [comment]
@keshakaneria That is intentional because that's how Django's ConsoleHandler works, if the right SMTP credentials are passed in, then it would use SMTP to send the email.

As for using celery, I don't find a strong reason as to why should we send one simple email using Celery. Feel free to prove me wrong.

**Comment by keshakaneria** [comment]
@SanketDG I wanted to use an 'Asynchronous task' where the mail will be sent in background and in the interface the according message will be displayed. Not using it will make the website keep loading until the mail through SMTP would send a mail and then return the next step.

**Comment by keshakaneria** [comment]
Sorry, by mistake I closed the issue. Its opened again.

**Comment by SanketDG** [comment]
> Not using it will make the website keep loading until the mail through SMTP would send and then return the next step.

That's fair, although sending one email would be mostly trivial, I am just not sure whether a highly cohesive dependency like Celery is worth it or not, most of the applications I have seen are for a large number of asynchronous tasks, that's where it shines.

This is definitely a valid issue though, let's see what other mentors think about this!

**Comment by keshakaneria** [comment]
Sure, thank you so much for letting me brainstorm again on this issue.

**Comment by satya7289** [comment]
@keshakaneria We can use cronjob.

**Comment by saideepesh000** [comment]
#OSH, Can I work on it.

**Comment by saideepesh000** [comment]
OSH, Can I work on it.

**Comment by saideepesh000** [comment]
@keshakaneria, can I use this [Password Reset View](https://docs.djangoproject.com/en/3.1/topics/auth/default/#django.contrib.auth.views.PasswordResetView), to reset the password without celery, as I don't have any experience using celery.

**Comment by keshakaneria** [comment]
@saideepesh000 Yes you can, unless and until it covers the main goal of the issue to send a mail and also displaying it in console to verify and reset the password successfully.

**Comment by saideepesh000** [comment]
@keshakaneria, Actually reset password section is already present on the website. and it's working.

![Screenshot (269)](https://user-images.githubusercontent.com/43727167/92106299-d12d3400-ee01-11ea-8784-10dee1c84ced.png)

**Comment by keshakaneria** [comment]
@saideepesh000 you dont have to include the code of its working. As you said the flow is absolutely perfect. What the aim of this issue to get the verification email on the mail id which is entered. Currently it is displayed in console, so in addition we need to get the mail in the id itself.

**Comment by saideepesh000** [comment]
@keshakaneria, Please once take a look at [this](https://drive.google.com/file/d/1b84v12d5ujF3NJoIH-LH8npKP2PyLmwm/view?usp=sharing),

**Comment by keshakaneria** [comment]
> @keshakaneria, Please once take a look at [this](https://drive.google.com/file/d/1b84v12d5ujF3NJoIH-LH8npKP2PyLmwm/view?usp=sharing),

You edited and added your code to send the verification mail right? Seems perfect to me. Great job 👍

**Comment by saideepesh000** [comment]
> 
> 
> > @keshakaneria, Please once take a look at [this](https://drive.google.com/file/d/1b84v12d5ujF3NJoIH-LH8npKP2PyLmwm/view?usp=sharing),
> 
> You edited and added your code to send the verification mail right? Seems perfect to me. Great job 👍

@keshakaneria, I just actually added only a few lines i.e to configure SMTP mail. Should I send a PR, so that you could understand it more?

**Comment by keshakaneria** [comment]
Yes @saideepesh000 you can.

**Comment by keshakaneria** [comment]
Any updates on PR?

**Comment by saideepesh000** [comment]
> @saideepesh000  Any updates on PR?

Actually I am busy with the placements going on, it'll be done in a day or two.

**Comment by keshakaneria** [comment]
> > @saideepesh000  Any updates on PR?
> 
> Actually I am busy with the placements going on, it'll be done in a day or two.

You shouldn't have taken up the issue i suppose! Let someone else work on this till that time so that our OSH issue doesnt get affected! You can prepare for the interviews until then. All the best!

**Comment by keshakaneria** [comment]
You can report me the updates done by you, else I will have to unassign you, since it has been 10 days this issue is assigned to you.

**Comment by saideepesh000** [comment]
Sorry for the inconvenience, I may not be working for a while, till then assign this issue to someone else, I'd joining you, after all my things got finished (professional and personal).

**Comment by keshakaneria** [comment]
> Sorry for the inconvenience, I may not be working for a while, till then assign this issue to someone else, I'd joining you, after all my things got finished (professional and personal).

Okay thank you. Is anyone else ready to work on this issue?

**Comment by kritikaparmar-programmer** [comment]
Hey @keshakaneria , I've worked with reset passwords in Django but with with celery..

**Comment by keshakaneria** [comment]
> Hey @keshakaneria , I've worked with reset passwords in Django but with with celery..

You can try your hand on this isue then. Go for it.

**Comment by kritikaparmar-programmer** [comment]
Hey @keshakaneria , I wanted to say not with celery..😅

**Comment by keshakaneria** [comment]
> Hey @keshakaneria , I wanted to say not with celery..😅

So in the work you've previously learn is reset password as you said. Do that fulfill our requirements of sending the mail to the id? Currently it is getting displayed in console so along with this, we need to send it to the mail too. You can read the mentioned issue I have tagged here in above comments, you will get an idea.

**Comment by kritikaparmar-programmer** [comment]
Actually we will just need the emailid and password of the gmail id from which we want to send email and then by doing some changes in settings.py we can easily send the email for password reset.. Then email will be  sent to user's id  which will have the link to change password..
I have worked like this.. @keshakaneria ..

**Comment by keshakaneria** [comment]
> Actually we will just need the emailid and password of the gmail id from which we want to send email and then by doing some changes in settings.py we can easily send the email for password reset.. Then email will be sent to user's id which will have the link to change password..
> I have worked like this.. @keshakaneria ..

That is also fine. I am saying that you do the changes according to the requirements and open a PR, then let me know the output of sending the mail to their inbox.

**Comment by priyanshurohilla** [comment]
@keshakaneria If this issue is open for reassignment, please assign it to me

**Comment by Aaishpra** [comment]
@keshakaneria can i try working on this?

**Comment by keshakaneria** [comment]
@Aaishpra Sure go for it.

**Comment by Aaishpra** [comment]
@keshakaneria just wanted to know if this issue has to be solved using celery only?

**Comment by keshakaneria** [comment]
> @keshakaneria just wanted to know if this issue has to be solved using celery only?

Yes @Aaishpra

**Comment by Aaishpra** [comment]
> > @keshakaneria just wanted to know if this issue has to be solved using celery only?
> 
> Yes @Aaishpra 

Thanks for clarifying 😄

**Comment by Aaishpra** [comment]
@keshakaneria Update: i am half way through the solution of this issue. But, since my mid semester exams are scheduled in a week so would be busy with them. Therefore i think i would require a bit more time to solve this and make the PR. Hope, you consider this.

**Comment by keshakaneria** [comment]
@Aaishpra I hope you are good to go now :)

**Comment by Aaishpra** [comment]
> @Aaishpra I hope you are good to go now :)

Aah, sorry for no updates@keshakaneria i am done with my exams and  will update you with the PR soon😅

**Comment by Aaishpra** [comment]
@keshakaneria well I tried lots of different approaches for solving this issue with celery (or even SMTP) but couldn't get the desired results. So, I am unassigning from this as I have already wasted a lot of your time, please make it available for others. Apologies again.

**Comment by keshakaneria** [comment]
> @keshakaneria well I tried lots of different approaches for solving this issue with celery (or even SMTP) but couldn't get the desired results. So, I am unassigning from this as I have already wasted a lot of your time, please make it available for others. Apologies again.

Ahh.. no worries! That is fine. I hope you've learnt alot while researching to work things out! Others issues are open you for you :)

**Comment by Aaishpra** [comment]
> > @keshakaneria well I tried lots of different approaches for solving this issue with celery (or even SMTP) but couldn't get the desired results. So, I am unassigning from this as I have already wasted a lot of your time, please make it available for others. Apologies again.
> 
> Ahh.. no worries! That is fine. I hope you've learnt alot while researching to work things out! Others issues are open you for you :)

Ya sure😊 i really learnt a lot.
